packages/askui-nodejs/src/core/cache/cache-manager.ts

```diff
@@ -0,0 +1,325 @@
+    };
+
+    if (parsed.version !== 1) {
+      throw new Error(`Unsupported cache file version: ${parsed.version}. Only version 1 is supported.`);
+    }
+    if (parsed.data === undefined) {
+      throw new Error('Data is required');
+    }
+    if (parsed.createAt === undefined) {
+      throw new Error('Create at is required');
+    }
+    if (parsed.version === undefined) {
+      throw new Error('Version is required');
+    }
+    if (parsed.type !== 'AskUI-Cache') {
+      throw new Error(`Unsupported cache file type: ${parsed.type}. Only 'AskUI-Cache' is supported.`);
+    }
+
+    this.createAt = new Date(parsed.createAt);
+    this.version = parsed.version;
+    this.type = parsed.type;
+    this.data = Object.keys(parsed.data).reduce((acc, validationType) => {
+      const nextAcc = { ...acc };
+      const validationEntries = parsed.data?.[validationType] ?? {};
+      nextAcc[validationType] = Object.keys(validationEntries).reduce((innerAcc, instruction) => {
+        const nextInnerAcc = { ...innerAcc };
+        const rawEntries = validationEntries[instruction] ?? [];
+        nextInnerAcc[instruction] = rawEntries
+          .map((entry) => CacheEntry.fromJson(entry))
+          .filter((entry: CacheEntry | undefined): entry is CacheEntry => entry !== undefined);
+        return nextInnerAcc;
+      }, {} as CacheDataByInstruction);
+      return nextAcc;
+    }, {} as CacheDataByValidationType);
+  }
+
+  loadFromFile(): void {
+    if (this.cacheFilePath === undefined) {
+      logger.debug('Cache file path is not set. Skipping cache load.');
+      return;
+    }
+
+    logger.debug(`Loading cache from file: ${this.cacheFilePath}`);
+    if (!fs.existsSync(this.cacheFilePath)) {
+      logger.warn(`Cache file does not exist: ${this.cacheFilePath}. Skipping cache load.`);
+      return;
+    }
+    const data = fs.readFileSync(this.cacheFilePath, 'utf8');
+    const json = JSON.parse(data);
+    if (json.version !== 1) {
+      throw new Error(`Unsupported cache file version: ${json.version}`);
+    }
+    this.loadFromFileVersion1(json);
+
+    this.relevantData = this.data[this.validationType] ?? {};
+  }
+
+  saveToFile(): void {
+    if (this.cacheFilePath === undefined) {
+      logger.debug('Cache file path is not set. Skipping cache save.');
+      return;
+    }
+
+    try {
+      logger.debug(`Saving cache to file: ${this.cacheFilePath}`);
+      const jsonObject = this.asJsonObject();
+      const jsonString = JSON.stringify(jsonObject, null, 2);
+      // create the directory if it doesn't exist
+      if (!fs.existsSync(path.dirname(this.cacheFilePath))) {
+        fs.mkdirSync(path.dirname(this.cacheFilePath), { recursive: true });
+      }
+      fs.writeFileSync(this.cacheFilePath, jsonString, 'utf8');
+      logger.debug(`Cache saved successfully to: ${this.cacheFilePath}`);
+    } catch (error) {
+      logger.error(`Error saving cache to file: ${error}. Skipping cache save.`);
+    }
```

Comment:
We should throw the error again. 

Or do you have a reason, why you don't want to fail?